examples/convert_mne_sample.py

```diff
@@ -29,17 +30,18 @@
 event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
             'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
 
-raw_file = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif')
+fname = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif')
 events_data = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw-eve.fif')
 output_path = op.join(data_path, '..', 'MNE-sample-data-bids')
 
 ###############################################################################
 # Finally, we specify the raw_file and events_data
 
-raw_to_bids(subject_id='01', run='01', session_id='01', task='audiovisual',
-            raw_file=raw_file, events_data=events_data,
-            output_path=output_path, event_id=event_id,
-            overwrite=True)
+raw = read_raw(fname)
+bids_fname = make_bids_filename(subject='01', run='01', session='01',
+                                task='audiovisual', suffix='meg.fif')
```

Comment:
maybe we don't need `kind` argument in `write_raw_bids` if we do this ...

Reply:
mmh why wouldn't we? The writing of some fields such as `EEGReference` is still contingent on the "kind" of data.

Reply:
Because it's in the "suffix" when you provide the `bids_fname` ...

Reply:
i would prefer to `kind` over suffix. Suffix implies to the user that if I add another file extension, MNE is able to say take a Neuroscan `.cnt` and output an `.edf` and that is something we are not capable of. We have worked on the underlying principle that you get the same file format out that you started with

Reply:
okay I could add a check so that the extension does not change? If you don't give the filename extension, you can't construct a full filename ...

Reply:
@teonbrooks how do you propose to resolve this? Should we have two arguments `kind` and `extension` ? Or do you say the extension can go in `suffix`?

Reply:
i can't get past explicitly specifying the modality in the suffix but it being determined automatically in the write function. i'm -1 on the way a filename is constructed.

Reply:
we don't have to determine the modality automatically if we provide it in the filename ...

Reply:
@teonbrooks do you have a suggestion for better API?

Reply:
I thought this through once more. So far, we have been using "suffix" in our internal API to construct filenames. See for instance [here](https://github.com/mne-tools/mne-bids/blob/master/mne_bids/mne_bids.py#L634), [here](https://github.com/mne-tools/mne-bids/blob/master/mne_bids/mne_bids.py#L658), and [here](https://github.com/mne-tools/mne-bids/blob/master/mne_bids/mne_bids.py#L662). My feeling is that we should not have different conventions in our internal API vs. our external API -- so I am actually okay leaving it as `suffix`. We can't have both `suffix` and `kind` to construct the last part of the filename because that would mean we have two ways of doing the same thing (goes against Zen of Python). #105 may be redundant after all or we could rewrite it to do some checks ...

Reply:
okay from my side.

Reply:
okay last commit should also take care of #105